[CI] TimeSeriesLifecycleActionsIT testWaitForSnapshot failing

**Build Scans:**
- [elasticsearch-periodic #11507 / openjdk25_checkpart5_java-matrix](https://gradle-enterprise.elastic.co/s/2wq7pegwc53kc)
- [elasticsearch-pull-request #106298 / part-5](https://gradle-enterprise.elastic.co/s/acyjoz3sg5xmy)
- [elasticsearch-pull-request #105811 / part-5](https://gradle-enterprise.elastic.co/s/ijuajyxuygcem)
- [elasticsearch-pull-request #105024 / part-5](https://gradle-enterprise.elastic.co/s/s5vtnm2tn4dse)
- [elasticsearch-pull-request #104986 / part-5](https://gradle-enterprise.elastic.co/s/4e7ns5lmcprde)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:ilm:qa:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT.testWaitForSnapshot" -Dtests.seed=5525928DA661DB2D -Dtests.locale=ak-Latn-GH -Dtests.timezone=Asia/Qyzylorda -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testWaitForSnapshot'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Index:{index=index-hxefxvqcxt, managed=true, policy=policy-DAZuI, index_creation_date_millis=1764164381926, time_since_index_creation=12.1s, lifecycle_date_millis=1764164381926, age=12.1s, age_in_millis=12102, phase=delete, phase_time_millis=1764164393138, action=wait_for_snapshot, action_time_millis=1764164383272, step=wait-for-snapshot, step_time_millis=1764164393138, is_auto_retryable_error=true, failed_step_retry_count=5, previous_step_info={type=illegal_state_exception, reason=configured policy 'kqqfuSsp' not found}, phase_execution={policy=policy-DAZuI, phase_definition={min_age=0ms, actions={wait_for_snapshot={policy=kqqfuSsp}}}, version=1, modified_date_in_millis=1764164382833}, skip=false}
SLM:{}
```

**Issue Reasons:**
- [main] 5 failures in test testWaitForSnapshot (0.6% fail rate in 841 executions)
- [main] 4 failures in step part-5 (0.8% fail rate in 497 executions)
- [main] 4 failures in pipeline elasticsearch-pull-request (0.8% fail rate in 511 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -240,9 +240,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.heap_attack.HeapAttackLookupJoinIT
   method: testLookupExplosionBigString
   issue: https://github.com/elastic/elasticsearch/issues/138510
-- class: org.elasticsearch.xpack.ilm.TimeSeriesLifecycleActionsIT
-  method: testWaitForSnapshot
-  issue: https://github.com/elastic/elasticsearch/issues/138669
 - class: org.elasticsearch.xpack.ml.integration.DatafeedJobsIT
   method: testDatafeedTimingStats_DatafeedRecreated
   issue: https://github.com/elastic/elasticsearch/issues/138348

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -243,9 +243,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.DatafeedJobsIT
   method: testDatafeedTimingStats_DatafeedRecreated
   issue: https://github.com/elastic/elasticsearch/issues/138348
-- class: org.elasticsearch.xpack.ilm.actions.DownsampleActionIT
-  method: testILMWaitsForTimeSeriesEndTimeToLapse
-  issue: https://github.com/elastic/elasticsearch/issues/138843
 - class: org.elasticsearch.xpack.ml.integration.DatafeedJobsIT
   method: testStopLookback_closeJobFalse
   issue: https://github.com/elastic/elasticsearch/issues/139024

--- a/x-pack/plugin/ilm/src/javaRestTest/java/org/elasticsearch/xpack/ilm/TimeSeriesLifecycleActionsIT.java
+++ b/x-pack/plugin/ilm/src/javaRestTest/java/org/elasticsearch/xpack/ilm/TimeSeriesLifecycleActionsIT.java
@@ -74,6 +74,7 @@ import static org.elasticsearch.xpack.TimeSeriesRestDriver.updatePolicy;
 import static org.elasticsearch.xpack.core.ilm.ShrinkIndexNameSupplier.SHRUNKEN_INDEX_PREFIX;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.not;
@@ -262,6 +263,7 @@ public class TimeSeriesLifecycleActionsIT extends IlmESRestTestCase {
         });
     }
 
+    @SuppressWarnings("unchecked")
     public void testWaitForSnapshot() throws Exception {
         createIndexWithSettings(
             client(),
@@ -282,7 +284,19 @@ public class TimeSeriesLifecycleActionsIT extends IlmESRestTestCase {
         assertBusy(() -> {
             Map<String, Object> indexILMState = explainIndex(client(), index);
             assertThat(indexILMState.get("action"), is("wait_for_snapshot"));
-            assertThat(indexILMState.get("failed_step"), is("wait-for-snapshot"));
+            if (indexILMState.containsKey("failed_step")) {
+                assertThat(indexILMState.get("failed_step"), is("wait-for-snapshot"));
+            } else {
+                // The failed step gets reset every time ILM retries, so we introduce an alternative check
+                // We check that ILM is re-trying the wait-for-snapshot step
+                assertThat(indexILMState.get("step"), is("wait-for-snapshot"));
+                assertThat(indexILMState.get(FAILED_STEP_RETRY_COUNT_FIELD), notNullValue());
+                assertThat((int) indexILMState.get(FAILED_STEP_RETRY_COUNT_FIELD), greaterThan(0));
+                // And that the previous step failed because the SLM policy was missing
+                assertThat(indexILMState.get("previous_step_info"), notNullValue());
+                Map<String, Object> previousStepInfo = (Map<String, Object>) indexILMState.get("previous_step_info");
+                assertThat(previousStepInfo.get("reason"), equalTo("configured policy '" + slmPolicy + "' not found"));
+            }
         }, slmPolicy);
         createSlmPolicy(slmPolicy, snapshotRepo); // put the slm policy back
         assertBusy(() -> {

